### ISSUE TITLE
Improved Spatial Class

### ISSUE THREAD

**Issue by vonflyhighace2** [godotengine/godot#7209]
**Operating system or device - Godot version:**
ANY. 
Future versions.

**Issue description** (what happened, and what was expected):
The special class really needs some love. The docs still aren't where the need to be, which means if your new, things can be a pain to figure out. You have to dig deep to get even the simplest of functions. I mean things like `get_transform().basis.x.x` just to get a simplified vector really speaks for its self. I'm not sure if this applies to the 2D portion of the engine, but that part seem to be further along than the 3D portion.

**Steps to reproduce:**

**Link to minimal example project** (optional but very welcome):

**Comment by bojidar-bg** [comment]
What about using `get_translation().x` directly?

**Comment by vonflyhighace2** [comment]
That would give me the x position but not the x normalized Vector. so if I wanted to figure out the forward/left/right of an object or compare its facing direction, usually its z, I couldn't with only getting its translations z parameter. Unity for example has directional transform vectors, like `transform.up` or `transform.down` which is separate from `transform.position.x` which in godot would be ` translation.x `or `translation.y`

**Comment by bojidar-bg** [comment]
@vonflyhighace2 What about using dot product or normalizing the whole vector? As in:
```gdscript
var norm_x = get_translation().normalized().x
var norm_xy = get_translation().normalized().dot(Vector3(sqrt(2), sqrt(2), 0))
```

About Transform.up or Vector3.up, I agree that those are useful, but they would need another issue. :smiley: 

Also, I still don't understand the exact thing you are trying to get by _manually_ inspecting the transformation _matrix_..

**Comment by bojidar-bg** [comment]
Ah, I think I get the issue now... You mean things like `Transform::{up,down,left,right,forward,backward}`, which are going to be normalized vector3-s?

**Comment by vonflyhighace2** [comment]
Yes, normalized vector 3's. Sorry if I didn't explain it in more details. It would be a good way to simplify getting the transform vector instead of the current way its done. Also the OpenGL convention of backwards is really forward on a transform, would be nice to have a more streamline fix since we're able to break things in the new 3.0 version.

**Comment by reduz** [comment]
Is this not Improved in 3.0 with direct property access?

**Comment by vonflyhighace2** [comment]
@reduz I have not taken 3.0 for a spine yet, still working on a project with 2.1.2 at the moment. Based on what bojidar-bg commented a day ago it seems the direct property access only accounted for object/node positions in the world { X, Y, Z} but not the transform direction of the object/node [ the forward axis of the camera for example, or if you wanted the direction a character was facing ], I could be wrong though, but if this has already been implemented then, you can ignore me :P.

**Comment by Noshyaar** [comment]
First of all thank you for your report and sorry for the delay.

We released Godot 3.0 in January 2018 after 18 months of work, fixing many old issues either directly, or by obsoleting/replacing the features they were referring to.

We still have hundreds of issues whose relevance/reproducibility needs to be checked against the [current stable version](https://godotengine.org/download), and that's where you can help us.
**Could you check if the issue that you described initially is still relevant/reproducible in Godot 3.0 or any newer version, and comment about it here?**

For bug reports, please also make sure that the issue contains detailed steps to reproduce the bug and, if possible, a zipped project that can be used to reproduce it right away. This greatly speeds up debugging and bugfixing tasks for our contributors.

Our Bugsquad will review this issue more in-depth in 15 days, and potentially close it if its relevance could not be confirmed.

Thanks in advance.

*Note: This message is being copy-pasted to many "stale" issues (90+ days without activity). It might happen that it is not meaningful for this specific issue or appears oblivious of the issue's context, if so please comment to notify the Bugsquad about it.*

**Comment by akien-mga** [comment]
As there was no answer to the above request, we're closing this issue as inactive.
If you can still reproduce it, please open a new issue against a currently supported release of Godot.
